### ISSUE TITLE
Can't join another manager or promote a worker

### ISSUE THREAD

**Issue by wyt** [moby/moby#38967]
* [ ] This is a bug report
* [ ] This is a feature request
* [x] I searched existing issues before opening this one

### Situation

``` bash
[root@dlink-ops-03-backend-204 swarm]# docker node ls
ID                            HOSTNAME                   STATUS              AVAILABILITY        MANAGER STATUS      ENGINE VERSION
bu9d3msrqh1m02172td16ity1 *   dlink-ops-03-backend-204   Ready               Active              Leader              18.09.0
x3cmymxu4619z7igzna4phj82     dlink-ops-04-backend-206   Ready               Active                                  18.09.0
345sr8grm673ki0ythjcm0a3w     dlink-ops-05-backend-209   Ready               Active                                  18.09.0
scelf9hji639qbgzuc128amw3     dlink-ops-06-backend-205   Ready               Active                                  18.09.0
[root@dlink-ops-03-backend-204 swarm]# docker node promote x3cmymxu4619z7igzna4phj82
Node x3cmymxu4619z7igzna4phj82 promoted to a manager in the swarm.
[root@dlink-ops-03-backend-204 swarm]# docker node ls
ID                            HOSTNAME                   STATUS              AVAILABILITY        MANAGER STATUS      ENGINE VERSION
bu9d3msrqh1m02172td16ity1 *   dlink-ops-03-backend-204   Ready               Active              Leader              18.09.0
x3cmymxu4619z7igzna4phj82     dlink-ops-04-backend-206   Down                Active                                  18.09.0
345sr8grm673ki0ythjcm0a3w     dlink-ops-05-backend-209   Ready               Active                                  18.09.0
scelf9hji639qbgzuc128amw3     dlink-ops-06-backend-205   Ready               Active                                  18.09.0
[root@dlink-ops-03-backend-204 swarm]# 
```
dlink-ops-04-backend-206 `docker info`
``` bash
[root@dlink-ops-04-backend-206 swarm]# docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 18.09.0
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: error
 NodeID: 
 Error: manager stopped: can't initialize raft node: rpc error: code = Unknown desc = could not connect to prospective new cluster member using its advertised address: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: authentication handshake failed: x509: certificate signed by unknown authority (possibly because of \"x509: ECDSA verification failure\" while trying to verify candidate authority certificate \"swarm-ca\")"
 Is Manager: false
 Node Address: 172.16.6.206
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: c4446665cb9c30056f4998ed953e6d4ff22c7c39
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-693.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 62.76GiB
Name: dlink-ops-04-backend-206
ID: 5DQ4:HE24:7QHC:Z27A:KFK3:OCSG:7F22:6YBA:AWGR:PVCN:7EHI:TSAO
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
HTTP Proxy: http://172.16.6.210:6666/
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 https://58mhm2vs.mirror.aliyuncs.com/
Live Restore Enabled: false
Product License: Community Engine
```

### Expected behavior
Promoted to management node

### Actual behavior

Promotion to management node failed

### Steps to reproduce the behavior

**Output of `docker version`:**
Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:19:08 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```
(paste your output here)
```

**Output of `docker info`:**
[root@dlink-ops-03-backend-204 swarm]# docker info
Containers: 15
 Running: 14
 Paused: 0
 Stopped: 1
Images: 14
Server Version: 18.09.0
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: active
 NodeID: bu9d3msrqh1m02172td16ity1
 Is Manager: true
 ClusterID: wqwl8cmok32u775f8y2h83nu2
 Managers: 1
 Nodes: 4
 Default Address Pool: 10.0.0.0/8  
 SubnetSize: 24
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 10
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 172.16.6.204
 Manager Addresses:
  172.16.6.204:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: c4446665cb9c30056f4998ed953e6d4ff22c7c39
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-693.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 62.76GiB
Name: dlink-ops-03-backend-204
ID: 77R4:HV5N:BFK6:NHSL:E5FY:HNXK:RAPP:SSHM:2WFP:V6WP:UBEM:W5FC
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
HTTP Proxy: http://172.16.6.210:6666/
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 https://58mhm2vs.mirror.aliyuncs.com/
Live Restore Enabled: false
Product License: Community Engine
```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.)**

**Comment by jaslinfernando** [comment]
I got the same problem. The error is '...Error: manager stopped: can't initialize raft node: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: authentication handshake failed: x509: certificate is not valid for any names, but wanted to match swarm-manager"...'

Thats error is present in node promote from the manager ($ docker node promote z126e55jym07zhz137494wuad) when I query ($ docker info) in the node worker.

How solve it?

**Comment by jaslinfernando** [comment]
I dit it!. The steps here.
I edit the file **$ sudo nano /etc/systemd/system/docker.service.d/http-proxy.conf**  and I set the **NO_PROXY** whit this: 
**Environment="NO_PROXY=localhost,127.0.0.1,172.16.23.117,172.16.23.124,172.16.23.116"** 

and restart thedaemon and docker

**$ systemctl daemon-reload
$ systemctl restart docker**


To add another manager node to the cluster in every one of managers, the **NO_PROXY** must be configured by adding the ip of each of the managers in each node.

That is, edit the **NO_PROXY** in each node manager. For example if the ip of the **host docker-17** is **172.16.23.117** and of the **host docker-24** is **172.16.23.124** and another of the **host docker-16** is **172.16.23.116** (3 managers in this case) the configuration in each  one of the manager nodes would be this: 

**Environment = "NO_PROXY = localhost, 127.0.0.1, 172.16.23.117,172.16.23.124,172.16.23.116"**

Then, in the new node manager run. 
**$ docker swarm join --token xyz 172.16.23.124:2377**

You must be note that mi first manager was the **host docker-24** and ip **172.16.23.124** and I run **$ docker swarm join --token xyz 172.16.23.124:2377** in the new node manager that I want to add.

That's it!

**Comment by thaJeztah** [comment]
What version of docker were you running?

**Comment by jaslinfernando** [comment]
Docker Engine 18.06.3-ce

**Comment by thaJeztah** [comment]
Looks like it's a duplicate of https://github.com/moby/moby/issues/35395 and https://github.com/moby/moby/issues/36951, which is related to proxy configuration, and should be fixed in Docker 18.09.4 and up through https://github.com/docker/swarmkit/pull/2802, which is vendor into master through https://github.com/moby/moby/pull/38831, and back ported to Docker 18.09 through https://github.com/docker/engine/pull/170

Closing this issue, but feel free to continue the conversation
